x-pack/legacy/plugins/lens/public/indexpattern_plugin/datapanel.tsx

```diff
@@ -102,34 +111,38 @@ export function IndexPatternDataPanel({
       timeFieldName: indexPatterns[id].timeFieldName,
     }));
 
-  useEffect(() => {
-    syncExistingFields({
-      dateRange,
-      setState,
-      indexPatterns: indexPatternList,
-      fetchJson: core.http.get,
-    });
-  }, [
-    dateRange.fromDate,
-    dateRange.toDate,
-    indexPatternList.map(x => `${x.title}:${x.timeFieldName}`).join(','),
-  ]);
-
   return (
-    <MemoizedDataPanel
-      currentIndexPatternId={currentIndexPatternId}
-      indexPatternRefs={indexPatternRefs}
-      indexPatterns={indexPatterns}
-      query={query}
-      dateRange={dateRange}
-      filters={filters}
-      dragDropContext={dragDropContext}
-      showEmptyFields={state.showEmptyFields}
-      onToggleEmptyFields={onToggleEmptyFields}
-      core={core}
-      onChangeIndexPattern={onChangeIndexPattern}
-      existingFields={state.existingFields}
-    />
+    <>
+      <Loader
+        load={() =>
```

Comment:
Now that this is extracted into a component, can this component create and manage an AbortController for us? I think it's a relatively small code change since the `fetch` function already accepts it.